### PR TITLE
[Email] Show notification when an email admin has a restrictive email view policy on Email Outbox and Sent Emails

### PR DESCRIPTION
**Pull request by darjoo**
#### Summary 
Due to default settings of email view policies, email admins are unable to view all emails. This fix shows a notification that route them to where they can fix the issue for their user.

image

#### Work Item(s) 
Fixes [AB#493661](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/493661)

### CODE REVIEW

File: src/System Application/App/Email/src/Email/EmailImpl.Codeunit.al [hunk 1]
```diff
@@ -650,6 +652,26 @@ codeunit 8900 "Email Impl"
             Error(InsufficientPermissionsErr);
     end;
 
+    procedure ShowAdminViewPolicyInEffectNotification()
+    var
+        EmailAccountImpl: Codeunit "Email Account Impl.";
+        AdminViewPolicyInEffectNotification: Notification;
+    begin
+        if (not EmailAccountImpl.IsUserEmailAdmin()) or (GetUserEmailViewPolicy() = Enum::"Email View Policy"::AllEmails) then
```
Comment (AndreasMoth) [inline src/System Application/App/Email/src/Email/EmailImpl.Codeunit.al]: Would be easier to read if you split this into two if statements

File: src/System Application/App/Email/src/Email/EmailImpl.Codeunit.al [hunk 2]
```diff
@@ -650,6 +652,26 @@ codeunit 8900 "Email Impl"
...
+            exit;
+
+        AdminViewPolicyInEffectNotification.Id := AdminViewPolicyInEffectNotificationIdTok;
+        AdminViewPolicyInEffectNotification.Message(AdminViewPolicyInEffectNotificationMsg);
+        AdminViewPolicyInEffectNotification.Scope := NotificationScope::LocalScope;
+        AdminViewPolicyInEffectNotification.AddAction('Update policy', Codeunit::"Email Impl", 'OpenEmailViewPoliciesPage');
```
Comment (AndreasMoth) [inline src/System Application/App/Email/src/Email/EmailImpl.Codeunit.al]: "Update policy" should be a label for translations

File: src/System Application/App/Email/src/Email/EmailImpl.Codeunit.al [hunk 3]
```diff
@@ -36,6 +36,8 @@ codeunit 8900 "Email Impl"
         EmailViewPolicyUsedTxt: Label 'Email View Policy is used', Locked = true;
         EmailViewPolicyDefaultTxt: Label 'Falling back to default email view policy: %1', Locked = true;
         EmailModifiedByEventTxt: Label 'Email has been modified by event', Locked = true;
+        AdminViewPolicyInEffectNotificationIdTok: Label '0ee5d5db-5763-4acf-9808-10905a8997d5', Locked = true;
+        AdminViewPolicyInEffectNotificationMsg: Label 'Your email view policy limits the emails visible. Please update your view policy to see all emails.';
```
Comment (AndreasMoth) [inline src/System Application/App/Email/src/Email/EmailImpl.Codeunit.al]: I'm not sure we should say "Please update your view policy", that sounds like something is setup incorrectly and we are asking them to fix it. Maybe just "You can update your view policy to see all emails."

**Review by JesperSchulz — APPROVED** [review]
I think this change will deflect some support cases. Approved.

**Review by AndreasMoth — APPROVED** [review]

**Pull request merged**